src/main/java/org/eclipsefoundation/marketplace/config/SecretConfigSource.java

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.config;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.ServiceLoader;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Reads in a set of secret configuration values from the secret.properties file
+ * in the resources folder. These values are only secret in that they are set to
+ * be ignored by Git and should be set on a per-server basis.
+ * 
+ * ConfigSource implementation was used to enable the usage of the
+ * {@link ServiceLoader} to load the configurations.
+ * 
+ * @author Martin Lowe
+ */
+public class SecretConfigSource implements ConfigSource {
+	private static final Logger LOGGER = LoggerFactory.getLogger(SecretConfigSource.class);
+
+	private Map<String, String> secrets;
+
+	@Override
+	public Map<String, String> getProperties() {
+		if (secrets == null) {
+			this.secrets = new HashMap<>();
+			// use the class loader to get the secret.properties file path
+			URL fURL = getClass().getClassLoader().getResource("secret.properties");
```

Comment:
Getting the secret from the classloader means that at one point you will package the secret along with binary. This is not a good security practice. The path to the secret properties file should also be a config. Use @Inject and document that this config should be pass as a system property at startup (-DsecretConfigPath=/path/to/secret)

Reply:
Because of the nature of the config using service loader, I don't have bean context so I can't use injection. I'll have to make do with `System.getProperty(name)` if I want to set that variable at runtime

Reply:
You're very right, it's begging to be a vector for leaking secret information. I've switched over locally to use the above call with a property name of `config.secret.path` to match the naming convention of other properties in the system.